src/go/rpk/pkg/cli/transform/logs.go

```diff
@@ -0,0 +1,525 @@
+/*
+* Copyright 2024 Redpanda Data, Inc.
+*
+* Use of this software is governed by the Business Source License
+* included in the file licenses/BSL.md
+*
+* As of the Change Date specified in that file, in accordance with
+* the Business Source License, use of this software will be governed
+* by the Apache License, Version 2.0
+ */
+
+package transform
+
+import (
+	"container/list"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"math"
+	"os"
+	"os/signal"
+	"regexp"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/kafka"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/twmb/franz-go/pkg/kadm"
+	"github.com/twmb/franz-go/pkg/kgo"
+	"go.uber.org/zap"
+)
+
+var (
+	minTime = time.Unix(0, 0).UTC()
+	maxTime = time.Unix(math.MaxInt64, math.MaxInt64).UTC()
+)
+
+func newLogsCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var follow bool
+	var head, tail int
+	var since, until timequery
+	since.time = minTime
+	until.time = maxTime
+	cmd := &cobra.Command{
+		Use:     "logs NAME",
+		Aliases: []string{"log"},
+		Short:   "View logs for a transform",
+		Long: `View logs for a transform.
+
+Data transform's STDOUT and STDERR are captured during runtime and written to 
+an internally managed topic _redpanda.transform_logs.
+This command ouputs logs for a single transform over a period of time and 
+printing them to STDOUT. The logs can be printed in various formats.
+
+The --follow flag allows for streaming logs continuously, by default only 
+emitted logs are displayed.
```

Comment:
@rockwotj What does "by default only emitted logs are displayed" mean -- are there any log entries that are left out? Does the --follow flag accept any other values?

Reply:
No `--follow` is a boolean flag. What I'm trying to say is that you only get logs that already exist by default. `--follow` will also wait and print new logs as come in realtime

Reply:
@rockwotj got it! In that case I'd go with mentioning the default behavior first, followed by what the flag changes. I'd probably phrase it like
```
By default, only logs that have been emitted are displayed. Use the --follow flag to stream new logs continuously.
```